worker_test.go

```diff
@@ -427,3 +427,118 @@ func TestWorkerMetricsForwardableMetrics(t *testing.T) {
 		})
 	}
 }
+
+func TestLocalWorkerSampleTimeseries(t *testing.T) {
+	w := NewWorker(1, true, nil, logrus.New(), nil)
+
+	m := samplers.UDPMetric{
+		MetricKey: samplers.MetricKey{
+			Name: "a.b.c",
+			Type: "histogram",
+		},
+		Digest: 1,
+		Scope:  samplers.LocalOnly,
+	}
+	m2 := samplers.UDPMetric{
+		MetricKey: samplers.MetricKey{
+			Name: "a.b.c",
+			Type: "counter",
+		},
+		Digest: 2,
+		Scope:  samplers.MixedScope,
+	}
+	w.SampleTimeseries(&m)
+	assert.Equal(t, uint64(1), w.uniqueMTS.Estimate())
+	w.SampleTimeseries(&m)
+	assert.Equal(t, uint64(1), w.uniqueMTS.Estimate())
+	w.SampleTimeseries(&m2)
+	assert.Equal(t, uint64(2), w.uniqueMTS.Estimate())
+}
+
+func TestLocalWorkerSampleForwardedTimeseries(t *testing.T) {
+	w := NewWorker(1, true, nil, logrus.New(), nil)
+
+	m := samplers.UDPMetric{
+		MetricKey: samplers.MetricKey{
+			Name: "a.b.c",
+			Type: "histogram",
+		},
+		Digest: 1,
+		Scope:  samplers.MixedScope,
+	}
+	m2 := samplers.UDPMetric{
+		MetricKey: samplers.MetricKey{
+			Name: "a.b.c",
+			Type: "counter",
+		},
+		Digest: 2,
+		Scope:  samplers.GlobalOnly,
+	}
+	w.SampleTimeseries(&m)
+	w.SampleTimeseries(&m2)
+	assert.Equal(t, uint64(0), w.uniqueMTS.Estimate())
+}
+
+func TestGlobalWorkerSampleTimeseries(t *testing.T) {
+	w := NewWorker(1, false, nil, logrus.New(), nil)
+
+	m := samplers.UDPMetric{
+		MetricKey: samplers.MetricKey{
+			Name: "a.b.c",
+			Type: "histogram",
+		},
+		Digest: 1,
+		Scope:  samplers.LocalOnly,
+	}
+	m2 := samplers.UDPMetric{
+		MetricKey: samplers.MetricKey{
+			Name: "a.b.c",
+			Type: "counter",
+		},
+		Digest: 2,
+		Scope:  samplers.GlobalOnly,
+	}
+	w.SampleTimeseries(&m)
+	w.SampleTimeseries(&m2)
+	assert.Equal(t, uint64(2), w.uniqueMTS.Estimate())
+}
+
+func BenchmarkPacketChanWork(b *testing.B) {
+	w := NewWorker(1, true, nil, logrus.New(), nil)
+	m := samplers.UDPMetric{
+		MetricKey: samplers.MetricKey{
+			Name: "counter",
+			Type: counterTypeName,
+		},
+		Value:      20.0,
+		Digest:     12345,
+		SampleRate: 1.0,
+		Scope:      samplers.MixedScope,
+	}
+
+	go w.Work()
+	b.ResetTimer()
+	for i := 0; i < b.N; i++ {
+		w.PacketChan <- m
+	}
+	w.Stop()
+}
+
+func BenchmarkSampleTimeseries(b *testing.B) {
+	w := NewWorker(1, true, nil, logrus.New(), nil)
+	m := samplers.UDPMetric{
```

Comment:
What we do in other benchmarks in Veneur (and I'd recommend here) is using an array with a number of different metrics (with different types), and using modular divison to quickly select the one to feed to `SampleTimeseries`. Because of branch prediction, always using the same metric type could artificially speed up the benchmark in a way that's not representative of real-world usage.